File: board_dsl/src/main/java/com/dooho/board/repository/BoardRepository.java
package com.dooho.board.repository;

import com.dooho.board.entity.BoardEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import org.springframework.transaction.annotation.Transactional;


import java.time.LocalDate;
import java.util.List;

@Repository
@Transactional
public interface BoardRepository extends JpaRepository<BoardEntity, Integer> {

    List<BoardEntity> findTop3ByBoardWriteDateAfterOrderByBoardLikeCountDesc(LocalDate boardWriteDate);

    List<BoardEntity> findByOrderByBoardWriteDateDescBoardNumberDesc();


    List<BoardEntity> findByBoardTitleContains(String boardTitle);

    List<BoardEntity> findByBoardWriterEmail(String userEmail);

    boolean existsByBoardTitle(String boardTitle);

    BoardEntity findByBoardNumber(Integer boardNumber);

    void deleteBoardEntityByBoardNumber(Integer BoardNumber);

    BoardEntity findByBoardTitle(String boardTitle);



}

File: board_spring/src/main/java/com/dooho/board/api/board/BoardRepository.kt
package com.dooho.board.api.board

import com.dooho.board.api.board.BoardEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import org.springframework.transaction.annotation.Transactional
import java.time.LocalDate

@Repository
@Transactional
interface BoardRepository : JpaRepository<BoardEntity?, Int?> {
    fun findTop3ByBoardWriteDateAfterOrderByBoardLikeCountDesc(boardWriteDate: LocalDate?): List<BoardEntity?>?
    fun findByOrderByBoardWriteDateDescBoardNumberDesc(): List<BoardEntity?>?
    fun findByBoardTitleContains(boardTitle: String?): List<BoardEntity?>?
    fun findByBoardWriterEmail(userEmail: String?): List<BoardEntity?>?
    fun existsByBoardTitle(boardTitle: String?): Boolean
    fun findByBoardNumber(boardNumber: Int?): BoardEntity?
    fun deleteBoardEntityByBoardNumber(BoardNumber: Int?)
}